File: swift/Sources/LibSignalClient/chat/SingleOutboundMessage.swift
//
// Copyright 2026 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

import Foundation

/// A message to send to a single device of a peer.
///
/// Used by APIs like ``UnauthMessagesService/sendMessage(to:timestamp:contents:auth:onlineOnly:urgent:)``
public struct SingleOutboundMessage<Contents> {
    public var deviceId: DeviceId
    public var registrationId: UInt32
    public var contents: Contents

    public init(deviceId: DeviceId, registrationId: UInt32, contents: Contents) {
        self.deviceId = deviceId
        self.registrationId = registrationId
        self.contents = contents
    }
}

public typealias SingleOutboundSealedSenderMessage = SingleOutboundMessage<Data>

File: node/ts/net/chat/SingleOutboundMessage.ts
//
// Copyright 2026 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

/**
 * A message to send to a single device of a peer.
 *
 * Used by APIs like `UnauthMessagesService.sendMessage`.
 */
export type SingleOutboundMessage<T> = Readonly<{
  deviceId: number;
  registrationId: number;
  contents: T;
}>;

export type SingleOutboundSealedSenderMessage = SingleOutboundMessage<
  Uint8Array<ArrayBuffer>
>;
